src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -34,133 +33,126 @@ bool ThetaStarPathPlanner::isUnBlocked(int row, int col)
+                                        Coordinate dest, double curr_to_new_dist)
 {
-    // Only process this GridCell if this is a valid one
-    if (isValid(pNew.first, pNew.second) == true)
+    // Only process this CellHeuristic if this is a valid one
+    if (isCoordValid(new_coord) == true)
     {
         // If the successor is already on the closed
         // list or if it is blocked, then ignore it.
         // Else do the following
-        if (closedList[pNew.first][pNew.second] == false &&
-            isUnBlocked(pNew.first, pNew.second) == true)
+        //
+        if (closed_list.find(new_coord) == closed_list.end() &&
+            isUnBlocked(new_coord) == true)
         {
-            double gNew;
-            int new_parent_i, new_parent_j;
-            int parent_i = cellDetails[pCurr.first][pCurr.second].parent_i_;
-            int parent_j = cellDetails[pCurr.first][pCurr.second].parent_j_;
-            if (lineOfSight(parent_i, parent_j, pNew))
+            double g_new;
+            Coordinate parent_new;
+            Coordinate parent =
+                cell_heuristics[current_coord.row()][current_coord.col()].parent;
+            if (hasLineOfSight(parent, new_coord))
             {
-                new_parent_i = parent_i;
-                new_parent_j = parent_j;
-                gNew         = cellDetails[parent_i][parent_j].g_ +
-                       calculateHValue(parent_i, parent_j, pNew);
+                parent_new = parent;
+                g_new      = cell_heuristics[parent.row()][parent.col()].g +
+                        calculateHValue(parent, new_coord);
             }
             else
             {
-                new_parent_i = pCurr.first;
-                new_parent_j = pCurr.second;
-                gNew = cellDetails[pCurr.first][pCurr.second].g_ + currToNextNodeDist;
+                parent_new = current_coord;
+                g_new      = cell_heuristics[current_coord.row()][current_coord.col()].g +
+                        curr_to_new_dist;
             }
 
-            double hNew = calculateHValue(pNew.first, pNew.second, dest);
-            double fNew = gNew + hNew;
+            double h_new = calculateHValue(new_coord, dest);
+            double f_new = g_new + h_new;
 
             // If it isn’t on the open list, add it to
             // the open list. Make the current square
             // the parent of this square. Record the
-            // f, g, and h costs of the square GridCell
+            // f, g, and h costs of the square CellHeuristic
             //			 OR
             // If it is on the open list already, check
             // to see if this path to that square is better,
             // using 'f' cost as the measure.
-            if (cellDetails[pNew.first][pNew.second].f_ == DBL_MAX ||
-                cellDetails[pNew.first][pNew.second].f_ > fNew)
+            if (cell_heuristics[new_coord.row()][new_coord.col()].f == DBL_MAX ||
```

Comment:
I was considering changing this to a map. I will investigate that

Reply:
To a map? Not quite sure that I follow. I'd also recommend against a map here, just because we do a _lot_ of lookups, and looking up a value at a known offset in a vector is much faster then doing the required hashing to lookup in a map.

Reply:
done